src/Database/Driver/Mysql.php

```diff
@@ -176,4 +176,18 @@ public function supportsNativeJson()
 
         return $this->_supportsNativeJson = version_compare($this->_version, '5.7.0', '>=');
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getConnectionId()
```

Comment:
Sorry, but I don't think querying connection id just for mysql is helpful here.

I'd rather see the query logger use the connection name. It is safe and easy to configure.

Reply:
Different connections may have the same connection name.

Reply:
F.ex. if you have a database transaction that executes several SQL statements and the same URL is requested at the same time from two or more different clients it is impossible to determine to what transaction (connection) belongs each SQL statement without logging the transaction id.

Reply:
I see.  I would suggest a client side id then.  This only works for mysql.

Reply:
I am not sure how the client id would help if the requests are send by two different tabs opened in the same browser for example.

Reply:
client-side for the db so cakephp.

Reply:
`getConnectionId()` can be overridden in driver classes. I believe that Postgres and SQLServer provide ways to get the connection id or something similar.